AsyncDisplayKit/Layout/ASDimension.mm

```diff
@@ -11,66 +11,243 @@
 {
-  if (type == ASRelativeDimensionTypePoints) {
+  if (type == ASDimensionTypePoints) {
     ASDisplayNodeCAssertPositiveReal(@"Points", value);
-  } else if (type == ASRelativeDimensionTypeFraction) {
+  } else if (type == ASDimensionTypeFraction) {
     // TODO: Enable this assertion for 2.0.  Check that there is no use case for using a larger value, e.g. to layout for a clipsToBounds = NO element.
-    // ASDisplayNodeCAssert( 0 <= value && value <= 1.0, @"ASRelativeDimension fraction value (%f) must be between 0 and 1.", value);
+    // ASDisplayNodeCAssert( 0 <= value && value <= 1.0, @"ASDimension fraction value (%f) must be between 0 and 1.", value);
   }
-  ASRelativeDimension dimension; dimension.type = type; dimension.value = value; return dimension;
+  ASDimension dimension; dimension.type = type; dimension.value = value; return dimension;
 }
 
-ASRelativeDimension ASRelativeDimensionMakeWithPoints(CGFloat points)
+ASOVERLOADABLE extern ASDimension ASDimensionMake(CGFloat points)
+{
+  return ASDimensionMake(ASDimensionTypePoints, points);
+}
+
+ASOVERLOADABLE ASDimension ASDimensionMake(NSString *dimension)
+{
+  // Handle empty string
+  if (dimension.length == 0) {
+    return ASDimensionMake(ASDimensionTypePoints, 0.0);
+  }
+  
+  // Handle points
+  NSUInteger pointsStringLocation = [dimension rangeOfString:@"pt"].location;
+  if (pointsStringLocation != NSNotFound) {
+    // Check if points is at the end and remove it
+    if (pointsStringLocation == (dimension.length-2)) {
+      dimension = [dimension substringToIndex:(dimension.length-2)];
+      return ASDimensionMake(ASDimensionTypePoints, dimension.floatValue);
+    }
+  }
+  
+  // Handle fraction
+  NSUInteger percentStringLocation = [dimension rangeOfString:@"%"].location;
+  if (percentStringLocation != NSNotFound) {
+    // Check if percent is at the end and remove it
+    if (percentStringLocation == (dimension.length-1)) {
+      dimension = [dimension substringToIndex:(dimension.length-1)];
+      return ASDimensionMake(ASDimensionTypeFraction, dimension.floatValue);
+    }
+  }
+
+  // Assert as parsing went wrong
+  ASDisplayNodeCAssert(NO, @"Parsing dimension failed");
+  return ASDimensionAuto;
+}
+
+ASDimension ASDimensionMakeWithPoints(CGFloat points)
 {
   ASDisplayNodeCAssertPositiveReal(@"Points", points);
-  return ASRelativeDimensionMake(ASRelativeDimensionTypePoints, points);
+  return ASDimensionMake(ASDimensionTypePoints, points);
 }
 
-ASRelativeDimension ASRelativeDimensionMakeWithFraction(CGFloat fraction)
+ASDimension ASDimensionMakeWithFraction(CGFloat fraction)
 {
-  // ASDisplayNodeCAssert( 0 <= fraction && fraction <= 1.0, @"ASRelativeDimension fraction value (%f) must be between 0 and 1.", fraction);
-  return ASRelativeDimensionMake(ASRelativeDimensionTypeFraction, fraction);
+  // ASDisplayNodeCAssert( 0 <= fraction && fraction <= 1.0, @"ASDimension fraction value (%f) must be between 0 and 1.", fraction);
```

Comment:
Same reason as we commented out a similar assert in `ASDimensionMake`. Pinterest had problems with it. Enabled it now though.